x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/utils.ts

```diff
@@ -206,3 +206,32 @@ export const transformOrImportError = (
     });
   }
 };
+
+export const getMapDuplicates = (
+  arr: Array<{ [key: string]: string }>,
+  prop: string
+): Map<string, number> =>
+  arr.reduce<Map<string, number>>((acc, item) => {
+    if (item[prop] != null && acc.has(item[prop])) {
+      const totalView = acc.get(item[prop]) ?? 1;
+      acc.set(item[prop], totalView + 1);
+    } else if (item[prop] != null) {
+      acc.set(item[prop], 1);
+    }
+    return acc;
+  }, new Map());
+
+export const getDuplicates = (someMap: Map<string, number>): string | null => {
+  const hasDuplicates = Array.from(someMap.values()).some(i => i > 1);
+  if (hasDuplicates) {
+    return Array.from(someMap.entries())
```

Comment:
Might be cleaner as `Object.keys(counts).filter(counts[key] > 1).join(', ')`

Reply:
Gave this a shot but couldn't get it to look much cleaner. Because this is a Map, the return by filter is a tuple of [key, value] which is not what I want to pass into the join.

Reply:
scratch that this turned out pretty nice. Thanks for the suggestion!